### PR TITLE
remove deprecated compute_on_step from torchmetrics/image

### PR DESCRIPTION
**Pull request by phaseolud**
## What does this PR do?

Part of #956  - image

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: torchmetrics/image/fid.py [hunk 1]
```diff
@@ -148,10 +148,6 @@ class FrechetInceptionDistance(Metric):
         is installed. Either install as ``pip install torchmetrics[image]`` or
         ``pip install torch-fidelity``
 
-    .. note:: the ``forward`` method can be used but ``compute_on_step`` is disabled by default (oppesit of
-        all other metrics) as this metric does not really make sense to calculate on a single batch. This
-        means that by default ``forward`` will just call ``update`` underneat.
```
Comment (Borda) [inline torchmetrics/image/fid.py:153]: I would keep it, just remove mention of ``compute_on_step``
cc: @SkafteNicki

Comment (justusschock) [reply torchmetrics/image/fid.py:153]: No, if you remove the mention it becomes meaningless IMO. I'd delete it

**Review by justusschock — APPROVED** [review]

**Comment by phaseolud** [comment]
Do I need to update the changelog as well?

**Comment by phaseolud** [comment]
> # [Codecov](https://codecov.io/gh/PyTorchLightning/metrics/pull/979?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PyTorchLightning) Report
> > Merging [#979](https://codecov.io/gh/PyTorchLightning/metrics/pull/979?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PyTorchLightning) ([775395b](https://github.com/PyTorchLightning/metrics/commit/775395b662679eb808e04197df62f6ee4304dd62)) into [master](https://codecov.io/gh/PyTorchLightning/metrics/commit/f8ef656efc022e0c2199811e1b3bd490bd0ba61d?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PyTorchLightning) ([f8ef656](https://github.com/PyTorchLightning/metrics/commit/f8ef656efc022e0c2199811e1b3bd490bd0ba61d)) will **decrease** coverage by `12%`.
> > The diff coverage is `100%`.
> 
> ```diff
> @@          Coverage Diff           @@
> ##           master   #979    +/-   ##
> ======================================
> - Coverage      95%    83%   -12%     
> ======================================
>   Files         177    177            
>   Lines        7513   7513            
> ======================================
> - Hits         7139   6255   -884     
> - Misses        374   1258   +884     
> ```

Maybe a simple question, but I'm trying to learn 😄: why does the coverage go down with this commit?

**Comment by phaseolud** [comment]
> # [Codecov](https://codecov.io/gh/PyTorchLightning/metrics/pull/979?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PyTorchLightning) Report
> > Merging [#979](https://codecov.io/gh/PyTorchLightning/metrics/pull/979?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PyTorchLightning) ([775395b](https://github.com/PyTorchLightning/metrics/commit/775395b662679eb808e04197df62f6ee4304dd62)) into [master](https://codecov.io/gh/PyTorchLightning/metrics/commit/f8ef656efc022e0c2199811e1b3bd490bd0ba61d?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PyTorchLightning) ([f8ef656](https://github.com/PyTorchLightning/metrics/commit/f8ef656efc022e0c2199811e1b3bd490bd0ba61d)) will **decrease** coverage by `12%`.
> > The diff coverage is `100%`.
> 
> ```diff
> @@          Coverage Diff           @@
> ##           master   #979    +/-   ##
> ======================================
> - Coverage      95%    83%   -12%     
> ======================================
>   Files         177    177            
>   Lines        7513   7513            
> ======================================
> - Hits         7139   6255   -884     
> - Misses        374   1258   +884     
> ```

Maybe a simple question, but why does the coverage go down so much with this commit?

File: torchmetrics/image/fid.py (hunk 1 above)
Comment (SkafteNicki) [reply torchmetrics/image/fid.py:153]: yes delete it

**Comment by SkafteNicki** [comment]
@phaseolud yes please add to this section:
https://github.com/PyTorchLightning/metrics/blob/a5f2e039f2d90402b402d8f18bfe8808dc6c75c6/CHANGELOG.md?plain=1#L36-L43

**Review by SkafteNicki — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Pull request merged**